src/MALA.jl

```diff
@@ -1,11 +1,13 @@
 struct MALA{D} <: MHSampler
     proposal::D
-
-    MALA{D}(proposal::D) where {D} = new{D}(proposal)
+    accepted_draws :: Int
```

Comment:
I don't remember the previous discussion but this design seems a bit surprising to me. I think the number of accepted and total draws is not a property of the sampler but rather of the resulting chain.

Reply:
Yeah, the thought here was that samplers have a record of success or failure. This would be useful for implementing delayed rejection or other sample massaging. But that information could probably be retrieved from the vector of transitions along the way. 

With that in mind, maybe we can stick with `Transition` objects just storing whether the sample was accepted or rejected and compute final stats in chain construction or in sampling if needed. I know this PR is already a mess, but I'll revert and make the adjustments.